src/AppInstallerCLICore/Commands/SearchCommand.cpp

```diff
@@ -66,6 +66,18 @@ namespace AppInstaller::CLI
         return "https://aka.ms/winget-command-search";
     }
 
+    void SearchCommand::ValidateArgumentsInternal(Args& execArgs) const
+    {
+        if (!execArgs.Contains(Args::Type::Query) &&
+            !execArgs.Contains(Args::Type::Id) &&
+            !execArgs.Contains(Args::Type::Name) &&
+            !execArgs.Contains(Args::Type::Moniker) &&
+            !execArgs.Contains(Args::Type::Tag) &&
+            !execArgs.Contains(Args::Type::Command))

```

Comment:
I am not sure if we want to support -n option here as there is a possibility user will specify a large number which may cause querying the entire catalog. Thoughts?

Also, if there are any other options, you'll want me to include/relax in install/search, let me know.

Reply:
I added a comment below regarding searching the entire catalog. I think "-n" should still be supported.